site/docs/v1/tech/apis/_application-request-body.adoc

```diff
@@ -501,6 +501,9 @@ include::../shared/_advanced-edition-blurb-api.adoc[]
 +
 :advanced_feature!:
 
+[field]#sourceApplicationId# [type]#[UUID]# [optional]#Optional# [since]#Available since 1.43.0#::
+The optional Id of an existing Application to make a copy of. A unique [field]#application.name# is required. If present, the [field]#application.id# value will used for the new Application. All other values will be copied from the source Application to the new Application.
```

Comment:
```suggestion
The optional Id of an existing Application from which a copy will be made.
+
A unique [field]#application.name# is required. If present, the [field]#application.id# value will used for the new Application. If no Id is specified, a secure random UUID will be generated and used.
+
All other values will be copied from the source Application to the new Application.
```

Reply:
Is this a deep copy? That is, will roles, `xmlSignatureLocation`, etc, all be copied and be independent of any changes to the source application?

Reply:
Thanks.  And good question.  It's mostly a deep copy, except the oauth configuration is cleared.  And checking on `xmlSignatureLocation` surfaced a bug with saml configuration.  https://github.com/FusionAuth/fusionauth-issues/issues/2118